src/include/nodes/memnodes.h

```diff
@@ -47,6 +47,9 @@ typedef struct MemoryContextMethods
 	Size		(*get_chunk_space) (MemoryContext context, void *pointer);
 	bool		(*is_empty) (MemoryContext context);
 	void		(*stats) (MemoryContext context, uint64 *nBlocks, uint64 *nChunks, uint64 *currentAvailable, uint64 *allAllocated, uint64 *allFreed, uint64 *maxHeld);
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+	void		(*save_top_chunk_stats)(MemoryContext context);
+#endif
```

Comment:
IMO it's worth to organize returning of chunks statistics in some internal structure (without printing) in `stats()` function. This doesn't spawn a new virtual function point especially with side effect inboard. All logging we then delegate to outer `MemoryContextStats*()` function